specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-05-31/consumption.json

```diff
@@ -120,115 +120,1382 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Consumption/reportconfigs": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "ReportConfig_List",
+        "description": "Lists all report configs for a subscription.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/consumption/"
+        },
+        "x-ms-examples": {
+          "ReportConfigList": {
+            "$ref": "./examples/ReportConfigList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReportConfigListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Consumption/reportconfigs": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "ReportConfig_ListByResourceGroupName",
+        "description": "Lists all report configs for a resource group under a subscription.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/consumption/"
+        },
+        "x-ms-examples": {
+          "ReportConfigList": {
+            "$ref": "./examples/ReportConfigList.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReportConfigListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Consumption/reportconfigs/{reportConfigName}": {
```

Comment:
Is reportConfigs created under a resource group or under a subscription?

Reply:
Shouldnt it be created under an RG

Reply:
We allow report configurations to be created under a subscription as well as under a RG. This determines scope of the data in the report. This is similar to what we do currently in Consumption Budgets as well.